tests/io/test_partitioned_dataset.py

```diff
@@ -274,17 +279,56 @@ def test_invalid_dataset_config(self, dataset_config, error_pattern):
     @pytest.mark.parametrize(
         "dataset_config",
         [
-            {"type": CSVDataSet, "versioned": True},
-            {"type": "pandas.CSVDataSet", "versioned": True},
+            {**dataset_config, "versioned": True}
+            for dataset_config in LOCAL_DATASET_DEFINITION
+            if isinstance(dataset_config, dict)
```

Comment:
Seems to be failing due to a Pylint bug (https://github.com/PyCQA/pylint/issues/2491, https://github.com/PyCQA/pylint/issues/3877).

```suggestion
            {**ds_config, "versioned": True}
            for ds_config in LOCAL_DATASET_DEFINITION
            if isinstance(ds_config, dict)
```